src/internal/connector/graph/service.go

```diff
@@ -307,9 +307,16 @@ func (handler *LoggingMiddleware) Intercept(
 		// special case for supportability: log all throttling cases.
 		if resp.StatusCode == http.StatusTooManyRequests {
 			logger.Ctx(ctx).Infow("graph api throttling", "method", req.Method, "url", req.URL)
-		}
-
-		if resp.StatusCode != http.StatusTooManyRequests && (resp.StatusCode/100) != 2 {
+		} else if resp.StatusCode/100 == 4 {
+			respDump, _ := httputil.DumpResponse(resp, true)
+			logger.Ctx(ctx).Infow(
+				"graph api error",
+				"status", resp.Status,
+				"method", req.Method,
+				"url", req.URL,
+				"response", string(respDump),
```

Comment:
Preference is to go the simpler route and dump the response on all non-2xx, non-throttling responses.